Sources/tart/Commands/Run.swift

```diff
@@ -31,48 +31,43 @@ struct Run: AsyncParsableCommand {
   func run() async throws {
     if recovery && vnc {
       print("You can't run in recovery and use VNC!")
-      Foundation.exit(1)      
+      Foundation.exit(1)
     }
     
     let vmDir = try VMStorageLocal().open(name)
     vm = try VM(vmDir: vmDir)
 
-    try await withThrowingTaskGroup(of: Void.self) { group in
-      group.addTask {
-        do {
-          try await vm!.run(recovery)
+    Task {
+      do {
+        try await vm!.run(recovery)
 
-          Foundation.exit(0)
-        } catch {
-          if error.localizedDescription.contains("Failed to lock auxiliary storage.") {
-            print("Virtual machine \"\(name)\" is already running!")
-          } else {
-            print(error)
-          }
-
-          Foundation.exit(1)
+        Foundation.exit(0)
+      } catch {
+        if error.localizedDescription.contains("Failed to lock auxiliary storage.") {
+          print("Virtual machine \"\(name)\" is already running!")
+        } else {
+          print(error)
         }
+
+        Foundation.exit(1)
       }
-      if vnc {
-        group.addTask(operation: {
-          do {
-            print("Waiting for the VM to boot...")
-            let resolvedIP = try await IP.resolveIP(vm!.config, secondsToWait: 60)
-            guard let ip = resolvedIP else {
-              throw IPNotFound()
-            }
-            let url = URL(string: "vnc://\(ip)")!
-            print("Opening \(url)")
-            NSWorkspace.shared.open(url)
-          } catch {
-            print("Failed to get an IP for screen sharing: \(error)")
-          }
-        })
-      } else if !noGraphics {
-        runUI()
+    }
+    if vnc {
+      do {
+        print("Waiting for the VM to boot...")
+        let resolvedIP = try await IP.resolveIP(vm!.config, secondsToWait: 60)
+        guard let ip = resolvedIP else {
+          throw IPNotFound()
+        }
+        let url = URL(string: "vnc://\(ip)")!
+        print("Opening \(url)")
+        NSWorkspace.shared.open(url)
+      } catch {
+        print("Failed to get an IP for screen sharing: \(error)")
       }
-
-      try await group.waitForAll()
+      vm?.wait()
```

Comment:
Should this be a `vm!` instead (see `vm!.run()` above)?

And it seems that waiting for the `VM.sema` will result in premature termination:

https://github.com/cirruslabs/tart/blob/0d633de04af2bb454d7296f358ff51eafc77b559/Sources/tart/VM.swift#L139-L150

We probably need another semaphore that gets unlocked after the cleanup is done.

Reply:
Hot sure if that will corrupt anything but how about [b98b46d](https://github.com/cirruslabs/tart/pull/129/commits/b98b46d471bbea1075ad0cf8c29a7b28fb4f1724) then?